[Flaking-Test] VPA  - pkg/admission-controller/certs_test.go

The recent [PR #7454](https://github.com/kubernetes/autoscaler/pull/7454) introduced a feature that allows the VPA admission controller to reload the caBundle certificate and patch the webhook. Since its merge, we've observed a flaky test that was added in the same PR.

The test TestUnchangedCAReloader intermittently fails with the following error:
```
E0212 15:12:52.541801   26749 certs.go:88] "Failed to reload certificate" err="open /tmp/TestKeypairReloader3869426187/001/cert.crt: no such file or directory"
--- FAIL: TestUnchangedCAReloader (1.73s)
    certs_test.go:384: 
        	Error Trace:	/home/runner/work/autoscaler/autoscaler/go/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller/certs_test.go:384
        	Error:      	Should be false
        	Test:       	TestUnchangedCAReloader
        	Messages:   	expected patch to not be called
    certs_test.go:393: 
        	Error Trace:	/home/runner/work/autoscaler/autoscaler/go/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller/certs_test.go:393
        	Error:      	Not equal: 
        	            	expected: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUU5akNDQXQ2Z0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFOTVFzd0NRWURWUVFLRXdKallUQWUKRncweU5UQXlNVEl4TlRFeU5UVmFGdzB5TnpBeU1USXhOVEV5TlRWYU1BMHhDekFKQmdOVkJBb1RBbU5oTUlJQwpJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBZzhBTUlJQ0NnS0NBZ0VBeCtvREFtQ1BmVEhnODZ1bzlBeWxRREhmCjJCVGJVbUI5a2FUSnNKV3p6bE9BTkg4OEorVkNpTGZrWXJlQUhpc1lxdDlYQkNhd0xRazNHSFJmWVdQdDB2VlAKNDFPZXhFaXBRdkY5SWdDcFo4ZGFUZlZCYVh2YnBkaUh2NVdoak5vWlVROUdyWjFKQzVnMlpuUmhTcjd4VE03QQpSMnRHYTZOc2dqY3ZOVk80dFRqTW0wSTdRWk85NXVILzR0cWNRZ0tUOVIzbVFZbkFCaTNaL3R6WE[45](https://github.com/kubernetes/autoscaler/actions/runs/13288454108/job/37102891947#step:8:46)cDQwZmU3CktoeU51MU9valM2QUlidVA4eGlyRjlJRFF5a0RveUY0SDdVbVNwNTRGOEtqMUNybnl1dlFyanZ2dFlFVFFYTUsKa1hDUWVSc21YWGN5aWQzRm00MlIwdWdRTFQ2bXB0VThiZ010VlZrbWxubVBGNnJna1ZldWZ6M1lXczFrMExLSwp3QkpFQ1lzZE1tdEVFOHVoeEx4ek5tUnByV1lYUFhjZFhQdmVPaFYwaTBIa3BKQkVMRmk2bW1pMWNqaWpuOGJQCllqZFJVSTZHejlVRUFkU2ZxaFR6d2oxNXkzTGJheHFJY3UzK0xXRjlLeEx3ZWxKYzNhaXpZbDlBQWlUN0NTWWcKYjFkYS9RSEQ2VWVEY3pKYWZRbDMwTEt3amxXZVp6MUh3UEhaYXRkZzJjZUt4NnFhZnB0NXBPamgwN0c3K2poNQphaDRUZFcyRDRvempKSHgzdTdXa0w4TGN5R1FoNEdWNzBjZ1V2SlNZSkxKd205cS9OTmhIMG5rWUVCV2RiVFVXCmRUbDVwNTBqdTI2T3ZXSnZDblZaK2RERE1jS0M1aFg5eDlycW9vc0NpKzA1NHBDRTY1Sy8vMWppOHNaTUo5NHEKK0xKSGpuRVZXSnhRL0RleHp3MENBd0VBQWFOaE1GOHdEZ1lEVlIwUEFRSC9CQVFEQWdLRU1CMEdBMVVkSlFRVwpNQlFHQ0NzR0FRVUZCd01DQmdnckJnRUZCUWNEQVRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUIwR0ExVWREZ1FXCkJCUkRnb1p6a1VaaWtiZWx3MTM3amVaU0pWNEhkVEFOQmdrcWhraUc5dzBCQVFzRkFBT0NBZ0VBaHVXWDlLTUIKb1lJeWhTWkpwcXpGSzE0NFVybE1MVjA3bjFaUVRKRXRLM3dCS3locGo0d2hNZ1g0TEJhMEtQUHlXTHNWRWE2eQpadlltd2RXMWFUQ0c1ckV3bjVkWER1TzBQUWdTM3ErTHRlNTdtTUJKMlU1UG02OSt0QVNOZWMyMjI2anViZFJ6Cll6TFdRTSt1Z1NRUWVLeldKbUVKSnd2ZTJqbUorcWxDKzlXMHAxbE8raktHdWQvV1pHYXUwS2d0eHFmU2xMamMKUm9vQU9yUnNVTFlpL3d3eitncFlpT1VrU3FRLzdTcEJuYTI2NGZreVRRVlp6d2pFUmx0VXA0Rm5ramZFY3NUUwpJMmxkZmUvLzlwcmM0bnN2THc5dVQrOGYvVmRBSk56VGhNeS9tRmhJMFZmeUZDY0w1bTRLc2c2aEh5cmxFaFgyCkFLNHd3NUpWYUZITnVWektDVjhOOG45NlFMUC9paGg4eW5Rb3Nwd3MvYU5DWnFWNmtxWURETWlDQmJyUEc1ak8KTFE0UXNmMzRTUVBDU05nTjRTYzYwV3RrcURuNmlpcXRoQlEyRlpPODNTajFWNFIrUVFGNk1vK1JFa3QwRTZDMwpncU9Xb1JhQWpGcFMvekl5WUlRaXE4WVJXY2dhSlF1Mko4dTNnaUF6bm1EYVIrL2Q3SGxkLzg4dG1qWDlRMjYrCnFZZENxeE5Yci9paXF6OWo1Z0h2QjhSZnRjV2tLMkE3aTV6NGo3WlpYdmFNdmlLWm83N1NRS1pFVXN4NDJ4WVMKUFBlWGFRYWVZb3FCZWlUOEJwOGpSTEUzYUxqalZIUnhGUk8wcFUxRlNDRnJWOVdtQWVtWEJIUXp0WFFoeVZtdAp6QVV2WDArUW91WS9WanFHbS82MDErMVVuL1AxZzVJaVRTZz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo="
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUU5akNDQXQ2Z0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFOTVFzd0NRWURWUVFLRXdKallUQWUKRncweU5UQXlNVEl4TlRFeU5UVmFGdzB5TnpBeU1USXhOVEV5TlRWYU1BMHhDekFKQmdOVkJBb1RBbU5oTUlJQwpJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBZzhBTUlJQ0NnS0NBZ0VBeCtvREFtQ1BmVEhnODZ1bzlBeWxRREhmCjJCVGJVbUI5a2FUSnNKV3p6bE9BTkg4OEorVkNpTGZrWXJlQUhpc1lxdDlYQkNhd0xRazNHSFJmWVdQdDB2VlAKNDFPZXhFaXBRdkY5SWdDcFo4ZGFUZlZCYVh2YnBkaUh2NVdoak5vWlVROUdyWjFKQzVnMlpuUmhTcjd4VE03QQpSMnRHYTZOc2dqY3ZOVk80dFRqTW0wSTdRWk85NXVILzR0cWNRZ0tUOVIzbVFZbkFCaTNaL3R6WE45cDQwZmU3CktoeU51MU9valM2QUlidVA4eGlyRjlJRFF5a0RveUY0SDdVbVNwNTRGOEtqMUNybnl1dlFyanZ2dFlFVFFYTUsKa1hDUWVSc21YWGN5aWQzRm00MlIwdWdRTFQ2bXB0VThiZ010VlZrbWxubVBGNnJna1ZldWZ6M1lXczFrMExLSwp3QkpFQ1lzZE1tdEVFOHVoeEx4ek5tUnByV1lYUFhjZFhQdmVPaFYwaTBIa3BKQkVMRmk2bW1pMWNqaWpuOGJQCllqZFJVSTZHejlVRUFkU2ZxaFR6d2oxNXkzTGJheHFJY3UzK0xXRjlLeEx3ZWxKYzNhaXpZbDlBQWlUN0NTWWcKYjFkYS9RSEQ2VWVEY3pKYWZRbDMwTEt3amxXZVp6MUh3UEhaYXRkZzJjZUt4NnFhZnB0NXBPamgwN0c3K2poNQphaDRUZFcyRDRvempKSHgzdTdXa0w4TGN5R1FoNEdWNzBjZ1V2SlNZSkxKd205cS9OTmhIMG5rWUVCV2RiVFVXCmRUbDVwNTBqdTI2T3ZXSnZDblZaK2RERE1jS0M1aFg5eDlycW9vc0NpKzA1NHBDRTY1Sy8vMWppOHNaTUo5NHEKK0xKSGpuRVZXSnhRL0RleHp3MENBd0VBQWFOaE1GOHdEZ1lEVlIwUEFRSC9CQVFEQWdLRU1CMEdBMVVkSlFRVwpNQlFHQ0NzR0FRVUZCd01DQmdnckJnRUZCUWNEQVRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUIwR0ExVWREZ1FXCkJCUkRnb1p6a1VaaWtiZWx3MTM3amVaU0pWNEhkVEFOQmdrcWhraUc5dzBCQVFzRkFBT0NBZ0VBaHVXWDlLTUIKb1lJeWhTWkpwcXpGSzE0NFVybE1MVjA3bjFaUVRKRXRLM3dCS3locGo0d2hNZ1g0TEJhMEtQUHlXTHNWRWE2eQpadlltd2RXMWFUQ0c1ckV3bjVkWER1TzBQUWdTM3ErTHRlNTdtTUJKMlU1UG02OSt0QVNOZWMyMjI2anViZFJ6Cll6TFdRTSt1Z1NRUWVLeldKbUVKSnd2ZTJqbUorcWxDKzlXMHAxbE8raktHdWQvV1pHYXUwS2d0eHFmU2xMamMKUm9vQU9yUnNVTFlpL3d3eitncFlpT1VrU3FRLzdTcEJuYTI2NGZreVRRVlp6d2pFUmx0VXA0Rm5ramZFY3NUUwpJMmxkZmUvLzlwcmM0bnN2THc5dVQrOGYvVmRBSk56VGhNeS9tRmhJMFZmeUZDY0w1bTRLc2c2aEh5cmxFaFgyCkFLNHd3NUpWYUZITnVWektDVjhOOG45NlFMUC9paGg4eW5Rb3Nwd3MvYU5DWnFWNmtxWURETWlDQmJyUEc1ak8KTFE0UXNmMzRTUVBDU05nTjRTYzYwV3RrcURuNmlpcXRoQlEyRlpPODNTajFWNFIrUVFGNk1vK1JFa3QwRTZDMwpncU9Xb1JhQWpGcFMvekl5WUlRaXE4WVJXY2dhSlF1Mko4dTNnaUF6bm1EYVIrL2Q3SGxkLzg4dG1qWDlRMjYrCnFZZENxeE5Yci9paXF6OWo1Z0h2QjhSZnRjV2tLMkE3aTV6NGo3WlpYdmFNdmlLWm83N1NRS1pFVXN4NDJ4WVMKUFBlWGFRYWVZb3FCZWlUOEJwOGpSTEUzYUxqalZIUnhGUk8wcFUxRlNDRnJWOVdtQWVtWEJIUXp0WFFoeVZtdAp6QVV2WDArUW91WS9WanFHbS82MDErMVVuL1AxZzVJaVRTZz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
        	            	+
        	Test:       	TestUnchangedCAReloader
        	Messages:   	expected CA to not change
E0212 15:12:57.508017   267[49](https://github.com/kubernetes/autoscaler/actions/runs/13288454108/job/37102891947#step:8:50) config.go:151] "Unable to parse webhook labels" err="labels 'foo,bar' are invalid, the format should be: 'key1:value1,key2:value2'"
FAIL
```

A failure from a recent run can be seen here: [GitHub Actions Run](https://github.com/kubernetes/autoscaler/actions/runs/13288454108/job/37102891947).

It appears the test is failing due to intermittent certificate reload issues and unexpected patch invocations. Further investigation is needed to determine the root cause and stabilize the test.

/area vertical-pod-autoscaler
/kind flake

Another similar flake: https://github.com/kubernetes/autoscaler/actions/runs/13314589637/job/37185375732?pr=7834
```
E0213 18:35:01.939167   25032 certs.go:88] "Failed to reload certificate" err="tls: private key does not match public key"
--- FAIL: TestUnchangedCAReloader (1.65s)
    certs_test.go:384: 
        	Error Trace:	/home/runner/work/autoscaler/autoscaler/go/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller/certs_test.go:384
        	Error:      	Should be false
        	Test:       	TestUnchangedCAReloader
        	Messages:   	expected patch to not be called
E0213 18:35:12.790110   2[50](https://github.com/kubernetes/autoscaler/actions/runs/13314589637/job/37185375732?pr=7834#step:8:51)32 config.go:151] "Unable to parse webhook labels" err="labels 'foo,bar' are invalid, the format should be: 'key1:value1,key2:value2'"
FAIL
FAIL	k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller	23.299s
```

This happened again: https://github.com/kubernetes/autoscaler/actions/runs/13444469376/job/37566506181?pr=7813

Similar flake to the previous one:
```
E0220 20:58:31.252767   24830 certs.go:88] "Failed to reload certificate" err="tls: failed to find any PEM data in certificate input"
--- FAIL: TestUnchangedCAReloader (3.49s)
    certs_test.go:384: 
        	Error Trace:	/home/runner/work/autoscaler/autoscaler/go/src/k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller/certs_test.go:384
        	Error:      	Should be false
        	Test:       	TestUnchangedCAReloader
        	Messages:   	expected patch to not be called
E0220 20:58:42.779159   2[48](https://github.com/kubernetes/autoscaler/actions/runs/13444469376/job/37566506181?pr=7813#step:8:49)30 config.go:151] "Unable to parse webhook labels" err="labels 'foo,bar' are invalid, the format should be: 'key1:value1,key2:value2'"
FAIL
```

--- a/vertical-pod-autoscaler/pkg/admission-controller/certs_test.go
+++ b/vertical-pod-autoscaler/pkg/admission-controller/certs_test.go
@@ -276,120 +276,119 @@ func TestChangedCAReloader(t *testing.T) {
 	assert.NotEqual(t, oldCAEncodedString, newCAEncodedString, "expected CA to change")
 }
 
-// TODO(omerap12): Temporary workaround for flakiness (#7831)
-// func TestUnchangedCAReloader(t *testing.T) {
-// 	tempDir := t.TempDir()
-// 	caCert := &x509.Certificate{
-// 		SerialNumber: big.NewInt(0),
-// 		Subject: pkix.Name{
-// 			Organization: []string{"ca"},
-// 		},
-// 		NotBefore:             time.Now(),
-// 		NotAfter:              time.Now().AddDate(2, 0, 0),
-// 		IsCA:                  true,
-// 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
-// 		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-// 		BasicConstraintsValid: true,
-// 	}
-// 	caKey, err := rsa.GenerateKey(rand.Reader, 4096)
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
-// 	caBytes, err := x509.CreateCertificate(rand.Reader, caCert, caCert, &caKey.PublicKey, caKey)
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
-// 	caPath := path.Join(tempDir, "ca.crt")
-// 	caFile, err := os.Create(caPath)
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
-// 	err = pem.Encode(caFile, &pem.Block{
-// 		Type:  "CERTIFICATE",
-// 		Bytes: caBytes,
-// 	})
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
+func TestUnchangedCAReloader(t *testing.T) {
+	tempDir := t.TempDir()
+	caCert := &x509.Certificate{
+		SerialNumber: big.NewInt(0),
+		Subject: pkix.Name{
+			Organization: []string{"ca"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(2, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+	caKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		t.Error(err)
+	}
+	caBytes, err := x509.CreateCertificate(rand.Reader, caCert, caCert, &caKey.PublicKey, caKey)
+	if err != nil {
+		t.Error(err)
+	}
+	caPath := path.Join(tempDir, "ca.crt")
+	caFile, err := os.Create(caPath)
+	if err != nil {
+		t.Error(err)
+	}
+	err = pem.Encode(caFile, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+	if err != nil {
+		t.Error(err)
+	}
 
-// 	testClientSet := fake.NewSimpleClientset()
+	testClientSet := fake.NewSimpleClientset()
 
-// 	selfRegistration(
-// 		testClientSet,
-// 		readFile(caPath),
-// 		0*time.Second,
-// 		"default",
-// 		"vpa-service",
-// 		"http://example.com/",
-// 		true,
-// 		int32(32),
-// 		"",
-// 		[]string{},
-// 		false,
-// 		"key1:value1,key2:value2",
-// 	)
+	selfRegistration(
+		testClientSet,
+		readFile(caPath),
+		0*time.Second,
+		"default",
+		"vpa-service",
+		"http://example.com/",
+		true,
+		int32(32),
+		"",
+		[]string{},
+		false,
+		"key1:value1,key2:value2",
+	)
 
-// 	webhookConfigInterface := testClientSet.AdmissionregistrationV1().MutatingWebhookConfigurations()
-// 	oldWebhookConfig, err := webhookConfigInterface.Get(context.TODO(), webhookConfigName, metav1.GetOptions{})
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
+	webhookConfigInterface := testClientSet.AdmissionregistrationV1().MutatingWebhookConfigurations()
+	oldWebhookConfig, err := webhookConfigInterface.Get(context.TODO(), webhookConfigName, metav1.GetOptions{})
+	if err != nil {
+		t.Error(err)
+	}
 
-// 	assert.Len(t, oldWebhookConfig.Webhooks, 1, "expected one webhook configuration")
-// 	webhook := oldWebhookConfig.Webhooks[0]
-// 	oldWebhookCABundle := webhook.ClientConfig.CABundle
+	assert.Len(t, oldWebhookConfig.Webhooks, 1, "expected one webhook configuration")
+	webhook := oldWebhookConfig.Webhooks[0]
+	oldWebhookCABundle := webhook.ClientConfig.CABundle
 
-// 	var reloadWebhookCACalled, patchCalled atomic.Bool
-// 	reloadWebhookCACalled.Store(false)
-// 	patchCalled.Store(false)
-// 	testClientSet.PrependReactor("get", "mutatingwebhookconfigurations", func(action k8stesting.Action) (bool, runtime.Object, error) {
-// 		reloadWebhookCACalled.Store(true)
-// 		return false, nil, nil
-// 	})
-// 	testClientSet.PrependReactor("patch", "mutatingwebhookconfigurations", func(action k8stesting.Action) (bool, runtime.Object, error) {
-// 		patchCalled.Store(true)
-// 		return false, nil, nil
-// 	})
+	var reloadWebhookCACalled, patchCalled atomic.Bool
+	reloadWebhookCACalled.Store(false)
+	patchCalled.Store(false)
+	testClientSet.PrependReactor("get", "mutatingwebhookconfigurations", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		reloadWebhookCACalled.Store(true)
+		return false, nil, nil
+	})
+	testClientSet.PrependReactor("patch", "mutatingwebhookconfigurations", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		patchCalled.Store(true)
+		return false, nil, nil
+	})
 
-// 	reloader := certReloader{
-// 		clientCaPath:          caPath,
-// 		mutatingWebhookClient: testClientSet.AdmissionregistrationV1().MutatingWebhookConfigurations(),
-// 	}
-// 	stop := make(chan struct{})
-// 	defer close(stop)
-// 	if err := reloader.start(stop); err != nil {
-// 		t.Error(err)
-// 	}
+	reloader := certReloader{
+		clientCaPath:          caPath,
+		mutatingWebhookClient: testClientSet.AdmissionregistrationV1().MutatingWebhookConfigurations(),
+	}
+	stop := make(chan struct{})
+	defer close(stop)
+	if err := reloader.start(stop); err != nil {
+		t.Error(err)
+	}
 
-// 	originalCaFile, err := os.ReadFile(caPath)
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
-// 	err = os.WriteFile(caPath, originalCaFile, 0666)
-// 	if err != nil {
-// 		t.Error(err)
-// 	}
+	originalCaFile, err := os.ReadFile(caPath)
+	if err != nil {
+		t.Error(err)
+	}
+	err = os.WriteFile(caPath, originalCaFile, 0666)
+	if err != nil {
+		t.Error(err)
+	}
 
-// 	oldCAEncodedString := base64.StdEncoding.EncodeToString(oldWebhookCABundle)
+	oldCAEncodedString := base64.StdEncoding.EncodeToString(oldWebhookCABundle)
 
-// 	for tries := 0; tries < 10; tries++ {
-// 		if reloadWebhookCACalled.Load() {
-// 			break
-// 		}
-// 		time.Sleep(1 * time.Second)
-// 	}
-// 	if !reloadWebhookCACalled.Load() {
-// 		t.Error("expected reloadWebhookCA to be called")
-// 	}
+	for tries := 0; tries < 10; tries++ {
+		if reloadWebhookCACalled.Load() {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	if !reloadWebhookCACalled.Load() {
+		t.Error("expected reloadWebhookCA to be called")
+	}
 
-// 	assert.False(t, patchCalled.Load(), "expected patch to not be called")
+	assert.False(t, patchCalled.Load(), "expected patch to not be called")
 
-// 	newWebhookConfig, err := webhookConfigInterface.Get(context.TODO(), webhookConfigName, metav1.GetOptions{})
-// 	assert.Nil(t, err, "expected no error")
-// 	assert.NotNil(t, newWebhookConfig, "expected webhook configuration")
-// 	assert.Len(t, newWebhookConfig.Webhooks, 1, "expected one webhook configuration")
+	newWebhookConfig, err := webhookConfigInterface.Get(context.TODO(), webhookConfigName, metav1.GetOptions{})
+	assert.Nil(t, err, "expected no error")
+	assert.NotNil(t, newWebhookConfig, "expected webhook configuration")
+	assert.Len(t, newWebhookConfig.Webhooks, 1, "expected one webhook configuration")
 
-// 	newWebhookCABundle := newWebhookConfig.Webhooks[0].ClientConfig.CABundle
-// 	newCAEncodedString := base64.StdEncoding.EncodeToString(newWebhookCABundle)
-// 	assert.Equal(t, oldCAEncodedString, newCAEncodedString, "expected CA to not change")
-// }
+	newWebhookCABundle := newWebhookConfig.Webhooks[0].ClientConfig.CABundle
+	newCAEncodedString := base64.StdEncoding.EncodeToString(newWebhookCABundle)
+	assert.Equal(t, oldCAEncodedString, newCAEncodedString, "expected CA to not change")
+}

